### PR TITLE
Feature/typedmesh

### PR DESCRIPTION
**Pull request by r-englund**
See #38

### CODE REVIEW

File: include/inviwo/core/datastructures/geometry/typedmesh.h [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+ * MyMesh mesh;
+ * mesh.addVertex(vec3(0.0f), vec4(1,0,0,1) );
+ * mesh.addVertex(vec3(1.0f), vec4(0,1,0,1) );
+ * \endcode
+ *
+ * If texture coordinates are also needed for each vertex in the mesh then one could instead use:
+ *
+ * \code{.cpp}
+ * using MyMesh = TypedMesh<buffertraits::PositionsBuffer, buffertraits::TexcoordBuffer,
+ * buffertraits::ColorsBuffer>; MyMesh mesh; mesh.addVertex(vec3(0.0f), vec3(0.0f), vec4(1,0,0,1) );
```
Comment (martinfalk) [inline include/inviwo/core/datastructures/geometry/typedmesh.h]: formatting issue

**Review by martinfalk — APPROVED** [review]
nice!

File: include/inviwo/core/datastructures/geometry/typedmesh.h [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+ */
+template <typename T, unsigned DIM, int attrib, int location = attrib>
+class TypedMeshBufferBase {
+public:
+    using type = Vector<DIM, T>;
+    static inviwo::Mesh::BufferInfo bi() {
+        return {static_cast<inviwo::BufferType>(attrib), location};
+    }
+
+#if _MSC_VER <= 1900
```
Comment (petersteneteg) [inline include/inviwo/core/datastructures/geometry/typedmesh.h]: if defined(_MSC_VER) && _MSC_VER <= 1900 ?

**Review by petersteneteg — APPROVED** [review]
Would be nice to get rid of the ifdef.... 

some comments on the actual functions in TypedMesh might also be nice

File: include/inviwo/core/datastructures/geometry/typedmesh.h (hunk 2 above)
Comment (martinfalk) [reply include/inviwo/core/datastructures/geometry/typedmesh.h]: can one not assume that _MSC_VER is given when the version number is there?

Comment (petersteneteg) [reply include/inviwo/core/datastructures/geometry/typedmesh.h]: Im not sure, think I remeber having issues with that before... that version is used below also...

Comment (r-englund) [reply include/inviwo/core/datastructures/geometry/typedmesh.h]: I htink Peter is right, if _MSC_VER is not defined the if statement will evaluate to 0 <= 1900

**Pull request merged**